impeller/compiler/compiler.cc

```diff
@@ -151,6 +160,7 @@ Compiler::Compiler(const fml::Mapping& source_mapping,
           shaderc_spirv_version::shaderc_spirv_version_1_0);
       break;
     case TargetPlatform::kFlutterSPIRV:
+    case TargetPlatform::kSkSL:  // TODO(zra): Allow optimizations.
```

Comment:
Let's file an issue for this and cross reference it here? Curious, did you try to see what happens? We can figure it out later though.

Reply:
The SkSL loop analysis code doesn't correctly handle everything that's allowed by the grammar. Added a comment, and filed an issue.